R/pkg-versions.R

```diff
@@ -0,0 +1,256 @@
+# Functions to extract version numbers of packages, both from CRAN and latest
+# GitHub releases.
+
+check_pkg_version_path <- function () {
+    cm_path <- normalizePath (file.path ("raw_cm.json"), mustWork = FALSE)
+
+    if (!file.exists (cm_path)) {
+        stop (
+            "This function can only be run in the ",
+            "output directory of 'roregistry', which ",
+            "must contain the 'raw_cm.json' file.",
+            call. = FALSE
+        )
+    }
+
+    return (cm_path)
+}
+
+#' Extract data on releases of packages, both on CRAN and GitHub.
+#'
+#' @param CRAN_only If `TRUE`, return only those rows of `repo_data` which
+#' correspond to packages which are or were on CRAN; otherwise return all data.
+#' @return A `data.frame` with one row for each package, and columns containing
+#' information on the package including dates and version numbers of GitHub and
+#' CRAN releases (if any).
+#'
+#' @export
+registry_pkg_versions <- function (CRAN_only = TRUE) {
+
+    cm_path <- check_pkg_version_path ()
+
+    # Code from list-packages.R to get all pkgs:
+    hosted_packages <- get_hosted_packages()
+    other_packages <- get_other_packages()
+    packages <- c(hosted_packages, other_packages)
+    packages <- packages[order(purrr::map_chr(packages, "package"))]
+    packages <- lapply (packages, function (i) i [c ("package", "url", "branch")])
```

Comment:
why not purrr? since it' used on the line above

Reply:
Yeah, no reason really, purrr throughout would be better